backend/shelf-function/src/main/java/com/htec/shelffunction/service/FunctionService.java

```diff
@@ -175,4 +178,77 @@ public void deleteFunction(Long functionId) {
 
         functionRepository.delete(functionEntity);
     }
+
+    public void createCustomFunction(CustomFunctionRequestModel customFunctionRequestModel, Long userId) {
+
+        FunctionEntity functionEntity = FunctionMapper.INSTANCE
+                .customFunctionRequestModelToFunctionEntity(customFunctionRequestModel);
+
+        functionEntity.setLanguage(customFunctionRequestModel.getLanguage());
+        functionEntity.setCustom(true);
+
+        functionRepository.saveAndFlush(functionEntity);
+
+        functionEntity.setPath(userId + pathSeparator + "functions" + pathSeparator + "Function" + functionEntity.getId());
+
+        functionRepository.save(functionEntity);
```

Comment:
why saving twice?

Reply:
The id is generated when entity is flushed to db,  then we need to add that id to path and save it second time